scripts/annotation_tool.py

```diff
@@ -0,0 +1,787 @@
+            no_of_models = len([os.path.basename(x)[:-4] for x in glob.glob(self.scenes.objects_path + '/*.ply')])
+            model_names = ['obj_' + f'{i + 1:06}' for i in range(no_of_models)]
+
+        return model_names
+
+    def _check_changes(self):
+        if self._annotation_changed:
+            self._on_error(
+                "Annotation changed but not saved. If you want to ignore the changes click the navigation button again.")
+            self._annotation_changed = False
+            return True
+        else:
+            return False
+
+    def _on_next_scene(self):
+        if self._check_changes():
+            return
+
+        if self._annotation_scene.scene_num + 1 > len(
+                next(os.walk(self.scenes.scenes_path))[1]):  # 1 for how many folder (dataset scenes) inside the path
+            self._on_error("There is no next scene.")
+            return
+        self.scene_load(self.scenes.scenes_path, self._annotation_scene.scene_num + 1,
+                        0)  # open next scene on the first image
+
+    def _on_previous_scene(self):
+        if self._check_changes():
+            return
+
+        if self._annotation_scene.scene_num - 1 < 1:
+            self._on_error("There is no scene number before scene 1.")
+            return
+        self.scene_load(self.scenes.scenes_path, self._annotation_scene.scene_num - 1,
+                        0)  # open next scene on the first image
+
+    def _on_next_image(self):
+        if self._check_changes():
+            return
+
+        num = len(
+            next(os.walk(os.path.join(self.scenes.scenes_path, f'{self._annotation_scene.scene_num:06}', 'depth')))[2])
+        if self._annotation_scene.image_num + 1 >= len(
+                next(os.walk(os.path.join(self.scenes.scenes_path, f'{self._annotation_scene.scene_num:06}', 'depth')))[
+                    2]):  # 2 for files which here are the how many depth images
+            self._on_error("There is no next image.")
+            return
+        self.scene_load(self.scenes.scenes_path, self._annotation_scene.scene_num, self._annotation_scene.image_num + 1)
+
+    def _on_previous_image(self):
+        if self._check_changes():
+            return
+
+        if self._annotation_scene.image_num - 1 < 0:
+            self._on_error("There is no image number before image 0.")
+            return
+        self.scene_load(self.scenes.scenes_path, self._annotation_scene.scene_num, self._annotation_scene.image_num - 1)
+
+
+def main():
+    parser = argparse.ArgumentParser(description="Manual annotation tool for BOP format")
+    # parser.add_argument("--dataset-path", type=str, help="dataset path", default='/media/gouda/all/datasets/HOPE')
+    # parser.add_argument("--dataset-split", type=str,
+    #                    help="dataset split to load (train[_TRAINTYPE], val[_VALTYPE], test[_TESTTYPE])",
+    #                    default='hope_val')
```

Comment:
remove please

Reply:
I forgot to totally remove the argprase part. I intended the script to have all the paths at the top (parameters) similar to the other scripts.